### PR TITLE
Improve documentation

### PR DESCRIPTION
**Pull request by skadauke**
# Description

This PR introduces another round of revisions and improvements to the documentation.

# Proposed Changes 

- Streamline and clarify roxygen2 docs for all exported functions. Make narrative style more uniform. Fix instances where code was not autolinked on the pkgdown site. Change from Rd to markdown wherever I make changes. 
- Add draft glossary vignette
- Update pkgdown YAML to highlight the glossary vignette and make small edits to the reference section
- Make nomenclature more consistent: change "(non)repeated" to "(non)repeating"; "tibble" to "data tibble" or "supertibble" where appropriate, "field" to "column" where appropriate, "option" to "choice"
- Revise the Getting Started and Diving Deeper vignettes
- Remove curlies or backticks around package names

### PR Checklist
Before submitting this PR, please check and verify below that the submission meets the below criteria:

- [x] Code changes are less than **250** lines total
- [x] The appropriate reviewer is assigned to the PR

# Code Review
This section to be used by the reviewer and developers during Code Review after PR submission

### Code Review Checklist

- [ ] I checked that new files follow naming conventions and are in the right place
- [ ] I checked that documentation is complete, clear, and without typos
- [ ] I added/edited comments to explain "why" not "how"
- [ ] I checked that all new variable and function names follow naming conventions
- [ ] I checked that new tests have been written for key business logic and/or bugs that this PR fixes
- [ ] I checked that new tests address important edge cases

### CODE REVIEW

File: R/read_redcap_tidy.R [hunk 1]
```diff
@@ -2,32 +2,32 @@
...
 #' data from all data collection instruments
-#' together. The `read_redcap_tidy()` function intelligently
+#' together. The `read_redcap_tidy()` function
 #' deconstructs the block matrix and splices the data into individual tibbles,
 #' where one tibble represents the data from one instrument.
 #'
 #' @return
-#' Returns a `tibble` in which each row represents a REDCap instrument. The
-#' results contains the following fields:
+#' Returns a `tibble` in which each row represents a REDCap instrument. It
```
Comment (ezraporter) [inline R/read_redcap_tidy.R:19]: Looks like elsewhere in the docs we leave the "returns" implicit so we should be consistent here
```suggestion
#' A `tibble` in which each row represents a REDCap instrument. It
```

**Review by ezraporter — APPROVED** [review]
Looks good! A couple minor tweaks to consider in the comments
